### PR TITLE
Refactor contributing

### PR DESCRIPTION
**Pull request by tjamir**

### CODE REVIEW

File: CI.MD [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Continuous Integration (CI)
+
+Our Continuous Integration pipeline currently runs on Github Actions. And is executed on push. As mentioned in [BUILDING.md](BUILDING.md), the whole process runs on a container. The following steps are followed throught a pipeline execution:
```
Comment (rodrigolc) [inline CI.MD:3]: throught -> through

File: README.md [hunk 2]
```diff
@@ -8,335 +8,81 @@ C extension for Spiffe platform.
...
-
-jobs:
-  # Set the job key. The key is displayed as the job name
-  # when a job name is not provided
-  super-lint:
-    # Name the Job
-    name: Lint code base
-    # Set the type of machine to run on
-    runs-on: ubuntu-latest
+[SPIFFE](https://spiffe.io/) stands for Seurity Identity Framewor for Evertyone and is a set for securely identifying system in dynamic and heteregeous environment. Please refer to https://spiffe.io/docs/latest/spiffe-about/overview/ for more information.
```
Comment (rodrigolc) [inline README.md:11]: too many typos

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+For building, we recommend to build an run inside a docker container. Refer to  [BUILDING.md](BUILDING.MD) for more information.
+
+## Conventions
+
+This project follows the [GNU Coding Standard](https://www.gnu.org/prep/standards/html_node/Writing-C.html) for formatting, commenting, and convetions in general. Our build system checks for formatting in the build proccess.
+
+## Continuous Integration
+
+This project uses Github Actions as Continuous Integration platform. Refer to [CI.md](CI.MD) for mor information.
```
Comment (rodrigolc) [inline CONTRIBUTING.md:24]: more information*

File: BUILDING.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+You can build the docker image from source with the above command line:
+
+```
+docker build -f docker/grpc.Dockerfile --build-arg GPRC_VERSION=1.34.0 --build-arg NUM_JOBS=8 --tag grpc-build:1.34.0 .
+```
+
+### Pull our Docker image
+
+Or you can use a pre build image:
```
Comment (rodrigolc) [inline BUILDING.md:20]: pre-built

File: BUILDING.md [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+## Docker Building(recommended)
+
+### Dependencies
+
+* Docker
+
+###  Build Docker Image
+
+You can build the docker image from source with the above command line:
```
Comment (rodrigolc) [inline BUILDING.md:12]: ... with the following command:

File: BUILDING.md [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+#### Building
+Build the c-spiffe project:
+```
+cd /mnt
+mkdir build && cd build
+cmake ..
+make
+make test
+```
+After running `make test`, you will find the test files into `Testing` folder.
```
Comment (rodrigolc) [inline BUILDING.md:54]: into -> in the

File: BUILDING.md [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+* lcov
+* gcovr
+* libjansson-dev
+* libcjose-dev
+* libgtest-dev
+* libgmock-dev
+
+### Compile GRPC
+
+In order to create a pure C lib, supporting GRPC, wich is a C++ lib, it is necessarty to compile GRPC code together with c-spiffe code. The following steps can be used for compiling GCC:
```
Comment (rodrigolc) [inline BUILDING.md:77]: wich -> which
GRPC-> gRPC
GCC -> gRPC

File: BUILDING.md [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+### Installing
+
+```
+make install
+```
+
+
+## Code Coverage Support
+
+This implements Code Coverage Reports using either using either `gcov` or `lcov`.
```
Comment (rodrigolc) [inline BUILDING.md:116]: This project generates Code Coverage Reports using `gcov` and `lcov`.

File: BUILDING.md [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+If you want to check them, you should run the following command after `make test`:
```
Comment (rodrigolc) [inline BUILDING.md:117]: check them -> generate coverage reports

File: BUILDING.md [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+
+```
+make gcov
+make lcov
+```
+
+The coverage reports will be into `Coverage` folder. In the case of `lcov`, you
```
Comment (rodrigolc) [inline BUILDING.md:124]: into -> saved in the

File: BUILDING.md [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+can see into the browser, opening the `index.html` file on the folder above.
```
Comment (rodrigolc) [inline BUILDING.md:125]: see into -> open the `index.html` file generated above in your browser.

File: CI.MD [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+* The project is checked out
+* Create a build environment and configure cmake
+* Build the project
+* Run unit tests
+* Generate coverage reports
+* Run integration tests
+* Upload coverage to code cov
+* Verify if current coverage is acceptable for lines and functions
+  
+If any of the above steps are failing, the build is broken. Any broken Pull Request should be rejected.
```
Comment (rodrigolc) [inline CI.MD:14]: "If any of the above steps fail, the build fails. Any Pull Request with failing builds should be rejected."

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+## Contributing
+
+In order to contribute to the project, the following steps should be followed:
+1. Fork this repository
+2. Create your feature branch
+3. Commit your local changes in the feature branch
+4. Push to your fork
+5. Create a pull Request
+
+Pull request are going to be reviewed by mainteners, some change can be requested before the Pull Request is accepted.
```
Comment (rodrigolc) [inline CONTRIBUTING.md:12]: ... maintainers, and changes can be ...

File: CONTRIBUTING.md [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+
+## Development setup
+
+For building, we recommend to build an run inside a docker container. Refer to  [BUILDING.md](BUILDING.MD) for more information.
```
Comment (rodrigolc) [inline CONTRIBUTING.md:16]: For development purposes, we recommend building and running the project inside...

File: CONTRIBUTING.md [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+
+## Conventions
+
+This project follows the [GNU Coding Standard](https://www.gnu.org/prep/standards/html_node/Writing-C.html) for formatting, commenting, and convetions in general. Our build system checks for formatting in the build proccess.
```
Comment (rodrigolc) [inline CONTRIBUTING.md:20]: conventions

File: CONTRIBUTING.md (hunk 3 above)
Comment (rodrigolc) [reply CONTRIBUTING.md:24]: ... Actions as its Continuous Integration tool. ( or solution? platform sounds wrong)

File: README.md [hunk 16]
```diff
@@ -8,335 +8,81 @@ C extension for Spiffe platform.
...
-jobs:
-  # Set the job key. The key is displayed as the job name
-  # when a job name is not provided
-  super-lint:
-    # Name the Job
-    name: Lint code base
-    # Set the type of machine to run on
-    runs-on: ubuntu-latest
+[SPIFFE](https://spiffe.io/) stands for Seurity Identity Framewor for Evertyone and is a set for securely identifying system in dynamic and heteregeous environment. Please refer to https://spiffe.io/docs/latest/spiffe-about/overview/ for more information.  
+C-spiffe is a extension for Spiffe that allows any [workload](https://spiffe.io/docs/latest/spiffe-about/spiffe-concepts/#workload) written C, C++ or even any language witch can import a .so library, access [Workload API](https://spiffe.io/docs/latest/spiffe-about/spiffe-concepts/#spiffe-workload-api) and establish a mTLS connection.
```
Comment (rodrigolc) [inline README.md:12]: written in* C, C++ or any language that supports loading a .so library, access the Workload API and establish mTLS connections with other workloads.

File: README.md [hunk 17]
```diff
@@ -8,335 +8,81 @@ C extension for Spiffe platform.
...
-  # Set the job key. The key is displayed as the job name
-  # when a job name is not provided
-  super-lint:
-    # Name the Job
-    name: Lint code base
-    # Set the type of machine to run on
-    runs-on: ubuntu-latest
+[SPIFFE](https://spiffe.io/) stands for Seurity Identity Framewor for Evertyone and is a set for securely identifying system in dynamic and heteregeous environment. Please refer to https://spiffe.io/docs/latest/spiffe-about/overview/ for more information.  
+C-spiffe is a extension for Spiffe that allows any [workload](https://spiffe.io/docs/latest/spiffe-about/spiffe-concepts/#workload) written C, C++ or even any language witch can import a .so library, access [Workload API](https://spiffe.io/docs/latest/spiffe-about/spiffe-concepts/#spiffe-workload-api) and establish a mTLS connection.  
+The image above shows a a example, where a C/C++ Workload imports the c-spiffe library in order to fetch a [SVID](https://spiffe.io/docs/latest/spiffe-about/spiffe-concepts/#spiffe-verifiable-identity-document-svid) and use it to establish a mTLS connection with another Workload, that can be implemented using another language, following Spiffe specification.
```
Comment (rodrigolc) [inline README.md:13]: which can be implemented in another language, provided it follows the SPIFFE standard.

File: README.md [hunk 18]
```diff
@@ -8,335 +8,81 @@ C extension for Spiffe platform.
...
-```
-</li>
-    <li>
-    To run your workflow, scroll to the bottom of the page and select Create a new branch for this commit and start a pull request. Then, to create a pull request, click Propose new file.
-    </li>
- </ol>
+### Motivation
 
- ![Alt text](img/commit-workflow-file.png)
+Even though there is an official [c-spiffe](https://github.com/spiffe/c-spiffe) library, we started this one from scratch. We wanted a C implementation (not C++) for better compatibility, and also we based all the design decisions on [go-spiffe](https://github.com/spiffe/c-spiffe), which is the offical supported extension by the Spiffe Community.
```
Comment (rodrigolc) [inline README.md:19]: and also based all...
official

File: README.md [hunk 19]
```diff
@@ -8,335 +8,81 @@ C extension for Spiffe platform.
...
+* **utils** Source code for utility functions
+* **workload** Source code for workload
+
+
+### Remarkable files
+
+* [CMakeList.txt](CMakeLists.txt) Main build configuration file. Each source folder also has its own CMake file.
+* [README.md](README.md) This README.
+* [LICENSE](LICENSE) Project license
+* [BUILDING.md](BUILDING.md) Instructions how to build c-spiffe in your system
```
Comment (rodrigolc) [inline README.md:53]: Instructions on how...

File: README.md [hunk 20]
```diff
@@ -8,335 +8,81 @@ C extension for Spiffe platform.
...
+* [CONTRIBUTING.md](CONTRIBUTING.md) Guide how to contribute c-spiffe project
```
Comment (rodrigolc) [inline README.md:54]: Guidelines for contributing to c-spiffe.

File: README.md [hunk 21]
```diff
@@ -8,335 +8,81 @@ C extension for Spiffe platform.
...
+## Using C-Spiffe
+
+### Installing
+
+#### Install from source
+Reffer to [BUILDING.md](BUILDING.md) 
+
+#### Install on system
+
+We are planning to deliver installers for main Linux distro, but at this moment, only building from source is supported.
```
Comment (rodrigolc) [inline README.md:67]: planning on delivering packages for the most popular Linux distros, ...

File: BUILDING.md (hunk 10 above)
Comment (rodrigolc) [reply BUILDING.md:124]: or "written to the"

File: BUILDING.md [hunk 22]
```diff
@@ -0,0 +1,125 @@
...
+cd /tmp/grpc/cmake/build && cmake -DgRPC_INSTALL=ON -DgRPC_BUILD_TESTS=ON ../..
+
+# Skipping benchmark tests
+RUN sed -i '7,13d' /tmp/grpc/third_party/benchmark/test/cxx03_test.cc 
+
+RUN cd /tmp/grpc/cmake/build && make -j${NUM_JOBS} && make install
+```
+### Compile C-Spiffe library
+
+Once you have all the requirements, building is straightfoward with `CMake`
```
Comment (rodrigolc) [inline BUILDING.md:98]: once you meet all....
straightforward

File: CI.MD (hunk 1 above)
Comment (rodrigolc) [reply CI.MD:3]: or "The following steps are run in order:"

**Review by rodrigolc — CHANGES_REQUESTED** [review]
minor fixes needed.

File: CONTRIBUTING.md (hunk 14 above)
Comment (arianalima) [inline CONTRIBUTING.md:16]:
```suggestion
For building, we recommend to build and run inside a docker container. Refer to [BUILDING.md](BUILDING.MD) for more information.
```

File: CI.MD (hunk 1 above)
Comment (otaviolcs3) [inline CI.MD:3]:
```suggestion
Our Continuous Integration pipeline currently runs on Github Actions and is executed on push. As mentioned in [BUILDING.md](BUILDING.md), the whole process runs on a container. The following steps are followed through a pipeline execution:
```

File: CONTRIBUTING.md [hunk 23]
```diff
@@ -0,0 +1,24 @@
+# Contributor guidelines
+
+## Contributing
+
+In order to contribute to the project, the following steps should be followed:
+1. Fork this repository
```
Comment (otaviolcs3) [inline CONTRIBUTING.md:6]:
```suggestion
1. Fork the main repository
```

File: CONTRIBUTING.md [hunk 24]
```diff
@@ -0,0 +1,24 @@
...
+2. Create your feature branch
+3. Commit your local changes in the feature branch
+4. Push to your fork
+5. Create a pull Request
```
Comment (otaviolcs3) [inline CONTRIBUTING.md:10]:
```suggestion
5. Create a pull request
```

File: CONTRIBUTING.md (hunk 13 above)
Comment (otaviolcs3) [inline CONTRIBUTING.md:12]:
```suggestion
Pull requests are going to be reviewed by maintainers, some change may be required before the Pull Request is accepted.
```

File: README.md [hunk 25]
```diff
@@ -8,340 +8,81 @@ C extension for Spiffe platform.
...
-```
-</li>
-    <li>
-    To run your workflow, scroll to the bottom of the page and select Create a new branch for this commit and start a pull request. Then, to create a pull request, click Propose new file.
-    </li>
- </ol>
+### Motivation
 
- ![Alt text](img/commit-workflow-file.png)
+Even though there is an official [c-spiffe](https://github.com/spiffe/c-spiffe) library, we started this one from scratch. We wanted a C implementation (not C++) for better compatibility, and also we based all the design decisions on [go-spiffe](https://github.com/spiffe/c-spiffe), which is the offical supported extension by the Spiffe Community.
```
Comment (otaviolcs3) [inline README.md:19]:
```suggestion
Even though there is an official [c-spiffe](https://github.com/spiffe/c-spiffe) library, we started this one from scratch. We wanted a C implementation (not C++) for better compatibility. We also based most of the design decisions on [go-spiffe](https://github.com/spiffe/c-spiffe), which is the official supported extension by the Spiffe Community.
```

**Review by otaviolcs3 — CHANGES_REQUESTED** [review]
Ver sugestões.

File: CONTRIBUTING.md [hunk 26]
```diff
@@ -0,0 +1,24 @@
...
+
+Pull requests are going to be reviewed by maintainers, some change may be required before the Pull Request is accepted.
+
+## Development setup
+
+For building, we recommend to build and run inside a docker container. Refer to [BUILDING.md](BUILDING.MD) for more information.
+
+## Conventions
+
+This project follows the [GNU Coding Standard](https://www.gnu.org/prep/standards/html_node/Writing-C.html) for formatting, commenting, and convetions in general. Our build system checks for formatting in the build proccess.
```
Comment (otaviolcs3) [inline CONTRIBUTING.md:20]:
```suggestion
This project follows the [GNU Coding Standard](https://www.gnu.org/prep/standards/html_node/Writing-C.html) for formatting, commenting, and conventions in general. Our build system checks for formatting in the build process.
```

File: BUILDING.md [hunk 27]
```diff
@@ -0,0 +1,126 @@
...
+
+```
+git clone git@github.com:HewlettPackard/c-spiffe.git
+cd c-spiffe
+docker build -f docker/grpc.Dockerfile --build-arg GPRC_VERSION=1.34.0 --build-arg NUM_JOBS=8 --tag grpc-build:1.34.0 .
+```
+
+### Pull our Docker image
+
+Or you can use a pre build image:
```
Comment (otaviolcs3) [inline BUILDING.md:21]:
```suggestion
Or you can use a pre-built image:
```

File: README.md [hunk 28]
```diff
@@ -8,340 +8,80 @@ C extension for Spiffe platform.
...
-
-jobs:
-  # Set the job key. The key is displayed as the job name
-  # when a job name is not provided
-  super-lint:
-    # Name the Job
-    name: Lint code base
-    # Set the type of machine to run on
-    runs-on: ubuntu-latest
+[SPIFFE](https://spiffe.io/) stands for Security Identity Framework for Everyone and is a set for securely identifying system in dynamic and heteregeneous environment. Please refer to https://spiffe.io/docs/latest/spiffe-about/overview/ for more information.
```
Comment (otaviolcs3) [inline README.md:11]:
```suggestion
[SPIFFE](https://spiffe.io/) stands for Security Identity Framework for Everyone and is a set for securely identifying system in dynamic and heterogeneous environment. Please refer to https://spiffe.io/docs/latest/spiffe-about/overview/ for more information.  
```

File: README.md [hunk 29]
```diff
@@ -8,340 +8,80 @@ C extension for Spiffe platform.
...
+* **utils** Source code for utility functions
+* **workload** Source code for workload
+
+
+### Remarkable files
+
+* [CMakeList.txt](CMakeLists.txt) Main build configuration file. Each source folder also has its own CMake file.
+* [README.md](README.md) This README.
+* [LICENSE](LICENSE) Project license
+* [BUILDING.md](BUILDING.md) Instructions how to build c-spiffe in your system
```
Comment (otaviolcs3) [inline README.md:53]:
```suggestion
* [BUILDING.md](BUILDING.md) Instructions on how to build c-spiffe in your system
```

File: README.md [hunk 30]
```diff
@@ -8,340 +8,79 @@ C extension for Spiffe platform.
...
-```
-After running `make test`, you will find the test files into `Testing` folder.
+### Remarkable files
 
-### Code Coverage Support
+* [CMakeList.txt](CMakeLists.txt) Main build configuration file. Each source folder also has its own CMake file.
+* [README.md](README.md) This README.
+* [LICENSE](LICENSE) Project license
+* [BUILDING.md](BUILDING.md) Instructions how to build c-spiffe in your system
+* [CONTRIBUTING.md](CONTRIBUTING.md) Guide how to contribute c-spiffe project
```
Comment (otaviolcs3) [inline README.md:54]:
```suggestion
* [CONTRIBUTING.md](CONTRIBUTING.md) Guide on how to contribute c-spiffe project
```

**Review by otaviolcs3 — APPROVED** [review]

**Review by rodrigolc — APPROVED** [review]

**Review by otaviolcs3 — APPROVED** [review]

**Pull request merged**